rust/toolchain.bzl

```diff
@@ -193,6 +193,7 @@ def _rust_toolchain_impl(ctx):
         rustfmt = ctx.file.rustfmt,
         cargo = ctx.file.cargo,
         clippy_driver = ctx.file.clippy_driver,
+        target_json = ctx.file.target_json,
```

Comment:
So there are multiple places where we compute the the value of the argument to the `--target` flag. How about we computed it only once here in the toolchain, and we added the string into the ToolchainInfo? I'd call it `target_flag_value`.

You'll still need a separate argument for the File instance so you can pass it as input to actions, or you can make the argument be part of the `rustc` depset. I prefer the latter.

Reply:
Done.